src/NovelRT/Graphics/GraphicsResourceManager.cpp

```diff
@@ -48,22 +71,51 @@ namespace NovelRT::Graphics
     GraphicsResourceManager& GraphicsResourceManager::operator=(GraphicsResourceManager&& other) noexcept
     {
         _graphicsDevice = std::move(other._graphicsDevice);
-        _stagingBuffer = Utilities::Lazy<std::shared_ptr<GraphicsBuffer>>([&]() { return CreateStagingBuffer(); });
-
-        if (other._stagingBuffer.isCreated())
-        {
-            _stagingBuffer.reset(other._stagingBuffer.getActual());
-            other._stagingBuffer.reset();
-        }
-
+        _stagingBuffers = std::move(other._stagingBuffers);
+        _defaultBuffers = std::move(other._defaultBuffers);
         _vertexBuffers = std::move(other._vertexBuffers);
+        _indexBuffers = std::move(other._indexBuffers);
+        _constantBuffers = std::move(other._constantBuffers);
         _textures = std::move(other._textures);
         _stagingBufferSize = other._stagingBufferSize;
         other._stagingBufferSize = 0;
+        _contextIndex = other._contextIndex;
+        other._contextIndex = 0;
 
         return *this;
     }
 
+    void GraphicsResourceManager::PrepForFrameWithContextIndex(size_t newContextIndex)
+    {
+        size_t finalContextCount = _contextIndex < newContextIndex ? newContextIndex : _contextIndex;
```

Comment:
What happens in the case of overflow here? I'm guessing overflow is an unlikely enough event that we have bigger issues in this case?

Reply:
This is out of date.

Reply:
Wait, actually, nevermind ignore previous comment - what overflow? It's always set by Vulkan.

Reply:
Mostly just curious on what happens if `newContextIndex` overflows from <some big value> to 0 - does Vulkan say that'll never happen?

Reply:
Vulkan will do whatever the GPU specifies in terms of minimum context count. Usually that's never more than 3 or 4.

Reply:
I see. I misread what the intent behind this variable was, and I was thinking it was an index.

Reply:
It is an index - but it is an index semantically tied to a graphics context, which there is never more than 3 or 4 of.